pkg/grpc/scheduler/client/client.go

```diff
@@ -1,18 +1,48 @@
 package client
 
 import (
+	"context"
+
+	scheduler_protos "github.com/square/p2/pkg/grpc/scheduler/protos"
 	"github.com/square/p2/pkg/manifest"
 	"github.com/square/p2/pkg/types"
 	"github.com/square/p2/pkg/util"
-
+	"google.golang.org/grpc"
 	klabels "k8s.io/kubernetes/pkg/labels"
 )
 
 type Client struct {
+	schedulerClient scheduler_protos.P2SchedulerClient
+}
+
+func NewClient(conn *grpc.ClientConn) Client {
+	return Client{
+		schedulerClient: scheduler_protos.NewP2SchedulerClient(conn),
+	}
 }
 
-func (c *Client) EligibleNodes(manifest.Manifest, klabels.Selector) ([]types.NodeName, error) {
-	return nil, util.Errorf("EligibleNodes() not yet implemented")
+func (c *Client) EligibleNodes(man manifest.Manifest, sel klabels.Selector) ([]types.NodeName, error) {
+	manifestStr, err := man.Marshal()
+	if err != nil {
+		return nil, util.Errorf("could not marshal manifest for EligibleNodes gRPC request: %s", err)
+	}
+
+	req := &scheduler_protos.EligibleNodesRequest{
+		Manifest:     string(manifestStr),
+		NodeSelector: sel.String(),
+	}
+
+	resp, err := c.schedulerClient.EligibleNodes(context.TODO(), req)
```

Comment:
I'm not an expert on contexts. Is context.TODO() something you anticipate replacing?

Reply:
it's the same thing as `context.Background()` except for there are some tools that can find `context.TOD()`. you're supposed to use it in places where the context might be threaded through someday, e.g. we add a `context.Context` argument to everything on the `Scheduler` interface. i'm not sure if we will do that because I don't think we'd ever want to cancel the gRPC, perhaps i should just just `context.Background()`